src/tree/tree.ts

```diff
@@ -109,22 +112,15 @@ export class SchemaTree extends Tree {
     if (node.children.length !== 0 || this.visited.has(node)) {
       return super.unwrap(node);
     }
-
     const metadata = getSchemaNodeMetadata(node);
     const { path, schemaNode, schema } = metadata;
     try {
-      if (!isRefNode(schemaNode)) {
+      if (!isRefNode(schemaNode) && !hasRefItems(schemaNode)) {
         this.populateTreeFragment(node, schema, path, true);
-      } else if (schemaNode.$ref !== null) {
-        const refPath = pointerToPath(schemaNode.$ref);
-        const schemaFragment = this.resolveRef
-          ? this.resolveRef(refPath, path, this.schema)
-          : _get(this.schema, refPath);
-        if (!_isObject(schemaFragment)) {
-          throw new ReferenceError(`Could not dereference "${pathToPointer(refPath)}"`);
-        }
-
-        this.populateTreeFragment(node, schemaFragment, path, false);
+      } else if ('$ref' in schemaNode && schemaNode.$ref) {
```

Comment:
```suggestion
      } else if (isRefNode(schemaNode)) {
```

Reply:
`isRefNode` checks only if `$ref` property is in node. It could be still falsy so I would rather make it:
``` } else if (isRefNode(schemaNode) && schemaNode.$ref) {```

or make `isRefNode` check if ref value is there

Reply:
> isRefNode checks only if $ref property is in node. It could be still falsy so I would rather make it:

that's okay if it falsy. We can error.